src/include/visibility.h

```diff
@@ -86,7 +86,7 @@
 #ifdef __cplusplus
 #include <cstdlib>
 #endif
-#pragma GCC poison exit
+//#pragma GCC poison exit
```

Comment:
I also have a question: I didn't figure out what to do about exit being poisoned but gtest using exit. Should I reenable it, remove visibility.h from the unit test builds? Is there an easy answer?

Reply:
This gets into how the binary is built. If we can get away with building separate test binaries or if the test itself needs to be an "app" within PX4.

My initial thought is that the options are
1.  build should be structured such that the unit test executable is not inheriting these flags at all 
2. each test is a module in PX4 and run like the existing unit tests. https://github.com/PX4/Firmware/blob/master/posix-configs/SITL/init/test/test_template.in#L24

Reply:
I would opt for 1. now and make sure to look into the possibilities of 2. as well at some point to be able to run the tests on the microcontroller targets.

Reply:
I figured that one one by removing the global `-include visibility.h` compile option in the gtest cmake. This flag should not be kept global in the future but rather added to the targets where it's necessary as discussed below.